docs/source/daml/reference/structure.rst

```diff
@@ -90,10 +90,26 @@ Here's the structure of a choice inside a template. There are two ways of specif
 
     Who can exercise the choice.
 
-:ref:`consumability <daml-ref-anytime>`
-    ``nonconsuming`` keyword
+:ref:`consumability <daml-ref-consumability>`
 
-    By default, contracts are archived when a choice on them is exercised, which means that choices can no longer be exercised on them. If you include ``nonconsuming``, this choice can be exercised over and over.
+    At most one of the following:
+
+    - ``preconsuming`` keyword
+
+        Exercising this choice archives the contract before the body is executed.
+        Contracts created in the choice body are known to the signatories and controllers of the contracts as well as to the observers of the contract on which the choice was exercised.
```

Comment:
See above wrt disclosure.

Reply:
But you mentioned divulgence above.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/4134/commits/9203ad8001c19412ac5692a2a726415320a891c9